src/sections/GiftRankingSection.tsx

```diff
@@ -130,9 +139,7 @@ export default function GiftRankingSection() {
   };
 
   const DEFAULT_VISIBLE_COUNT = 6;
-  const MAX_VISIBLE_COUNT = 12;
-  const repeatedData = Array(12).fill(null).flatMap(() => giftRankingData).slice(0, MAX_VISIBLE_COUNT);  
-  const visibleData = showAll ? repeatedData : repeatedData.slice(0, DEFAULT_VISIBLE_COUNT);
+  const visibleData = showAll ? data ?? [] : (data ?? []).slice(0, DEFAULT_VISIBLE_COUNT);
```

Comment:
```suggestion
  const visibleProducts = data ?? [];
  const productToShow = showAll ? visibleItems : visibleItems.slice(0, DEFAULT_VISIBLE_COUNT);
```

어.. 요건 동작은 맞는데 읽는 사람이 조금 어려울 수도 있을 거 같습니다.
코드가 조금 길어지더라도 위와 같이 표현해주면 조금 더 순서대로 읽을 수 있을 거 같아요!

보통 이렇게 조건문과 여러 연산자가 중첩된 상황을 인지복잡도라고 하는데요.
부서마다 이 인지복잡도의 제한을 걸어놓는 부서도 있습니다!

코드는 동료와 함께 짜는 것이니까요!

Reply:
이런 생각은 못해봤습니다... 아직은 혼자 코드를 작성하는게 익숙해서요... 다음부터는 신경써서 해보겠습니다!